### ISSUE TITLE
filtering genes post-seurat obj creation

### ISSUE THREAD

**Issue by brianjohnhaas** [satijalab/seurat#147]
Is there a way to filter genes after having created the Seurat object?

In the CreateSeuratObject function, there's:

```R
>  # filter genes on the number of cells expressing
>   # modifies the raw.data slot as well now
>   genes.use <- rownames(object@data)
>   if (min.cells > 0) {
>     num.cells <- rowSums(object@data > 0)
>     genes.use <- names(num.cells[which(num.cells >= min.cells)])
>     object@raw.data <- object@raw.data[genes.use, ]
>     object@data <- object@data[genes.use, ]
>   }
```

and, similarly to filtering based on cells, it would be useful to be able to perform these operations during subsequent QC steps instead (or in addition to) the initial object creation.

**Comment by satijalab** [comment]
Agreed - we'll fix in a future release

**Comment by graddi** [comment]
Hello! Any update on whether this function was introduced? Thanks for all your work.

**Comment by satijalab** [comment]
We looked into this further and decided that gene filtration could be misleading, as that could throw off normalization assumptions, etc.

We suggest that you filter genes in the original expression matrix, prior to constructing the Seurat object.

**Comment by rahulnutron** [comment]
I want to filter cells after obtaining clusters. This is because, in my case for few cells, some important genes are not expressed (but they are in the same cluster with other cells which have important genes), so I don't want them to be look for DEGs while running FindMarkers. But when I tried to filter the cells using ID

FilterCells(a.combined, subset.names='rep1_AAACCTGAGTGTGAAT')

I got error as 

Error in FetchData(object = object, vars.all = subset.name, cells.use = cells.use,  : 
  Error: rep1_AAACCTGAGTGTGAAT not found

Is there a way to remove those cells which are not important for comparison for DEGs after forming clusters?
